lib/readAccessor.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+var Cesium = require('cesium');
+var Cartesian2 = Cesium.Cartesian2;
+var Cartesian3 = Cesium.Cartesian3;
+var Cartesian4 = Cesium.Cartesian4;
+var Matrix2 = Cesium.Matrix2;
+var Matrix3 = Cesium.Matrix3;
+var Matrix4 = Cesium.Matrix4;
+var byteLengthForComponentType = require('./byteLengthForComponentType');
+var getAccessorByteStride = require('./getAccessorByteStride');
+var numberOfComponentsForType = require('./numberOfComponentsForType');
+
+module.exports = readAccessor;
+
+// read the accessor shown and return an array containing its contents
+// VECx types become CartesianX
+// MATx types become MatrixX
+function readAccessor(accessor, gltf) {
+    var bufferView = gltf.bufferViews[accessor.bufferView];
+    var buffer = gltf.buffers[bufferView.buffer];
+    var source = buffer.extras._pipeline.source;
+    var attributeCount = accessor.count;
+    var componentCount = numberOfComponentsForType(accessor.type);
+    var componentBytes = byteLengthForComponentType(accessor.componentType);
+    var attributeStride = componentCount * componentBytes;
+    var accessorType = accessor.type;
+    var componentType = accessor.componentType;
+    var attributes = [];
+    var currentBufferIndex = accessor.byteOffset + bufferView.byteOffset;
+
+    var readFunction;
+    switch (componentType) {
```

Comment:
See [readBufferComponentType](https://github.com/AnalyticalGraphicsInc/gltf-pipeline/blob/master/lib/readBufferComponentType.js). Should simplify this.